codec.go

```diff
@@ -157,6 +160,23 @@ func WithDecodeOnly() Option {
 	})
 }
 
+// WithCustomHeaders adds custom header to append to the http headers of the request sent to LargePayloadService
+// customHeaders needs to be initialised with a non-empty map and cannot contain nil values.
+func WithCustomHeaders(customHeaders map[string][]string) Option {
```

Comment:
Would it be nicer to have `WithCustomHeader(key string, value string)` only and let the user call it twice if he needs to set two values for the same key? This API you need to create a map and string slice to set a single key value pair.

Reply:
your point makes sense, I updated the code.